### DISCUSSION TITLE
Boolean false as payload for dispatchBrowserEvent results in empty object

### DISCUSSION THREAD

**Discussion by emilsrits** [livewire/livewire#3551] [7. Bugs (archived)]
Livewire version: 2.5

Passing `false` as payload in `dispatchBrowserEvent('eventName', $payload)` results in empty object `{}` on JavaScript side as opposed to boolean `false`.

```php
$this->dispatchBrowserEvent('foo', false);
```
```js
window.addEventListener('foo', (e) => {
    console.log(e.detail); // results in empty object - {}
});
```
---
```php
$this->dispatchBrowserEvent('foo', true);
```
```js
window.addEventListener('foo', (e) => {
    console.log(e.detail); // results in boolean - true
});
```

EDIT:
Passing zero `0` also returns empty object on JavaScript side.

**Comment by joshhanley** [comment]
@emilsrits hmm that's not ideal.

A work around for now could be to use an associative array.

```php
$this->dispatchBrowserEvent('foo', ['foo' => false]);
```
```js
window.addEventListener('foo', (e) => {
    console.log(e.detail.foo); // hopefully returns false??
});
```

Try that and see if it works, hopefully it should.

Hope this helps!
